### ISSUE TITLE
Segfault in verifying data after rolling upgrade.

### ISSUE THREAD

**Issue by amoskong** [scylladb/scylla#4031]
*Installation details*
Scylla version (or git commit hash): 3.0.rc3-0.20181224.5558fa8c4
Cluster size: 4 
OS (RHEL/CentOS/Ubuntu/AWS AMI): CentOS

* The segfault occurred after rolling upgrade (all ndoes were upgraded from 2.3.1-0.20181021.336c77166 to 3.0.rc3
* sst3 is enabled
* db log after segfault: `scylla-server.service holdoff time over, scheduling restart.`
* query around the problem occurred: `SELECT * FROM ks_no_range_ghost_test.users`
   test code: https://github.com/scylladb/scylla-cluster-tests/blob/master/sdcm/fill_db_data.py#L912
* only occurred once with this version
* I'm testing with latest 3.0.rc3 to know if it can be reproduced.
* db log: [segfault-db-log.txt](https://github.com/scylladb/scylla/files/2715070/segfault-db-log.txt)

```
[Backtrace #0]
void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at /usr/src/debug/scylla-3.0.rc3/seastar/util/backtrace.hh:56
seastar::backtrace_buffer::append_backtrace() at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:410
 (inlined by) print_with_backtrace at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:431
print_with_backtrace at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:438
sigsegv_action at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:4020
 (inlined by) operator() at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:4006
 (inlined by) _FUN at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:4002
_L_unlock_13 at funlockfile.c:?
blob_storage::ref_type::operator blob_storage*() const at /usr/src/debug/scylla-3.0.rc3/utils/managed_bytes.hh:40
 (inlined by) managed_bytes::~managed_bytes() at /usr/src/debug/scylla-3.0.rc3/utils/managed_bytes.hh:225
 (inlined by) compound_wrapper<clustering_key_prefix, clustering_key_prefix_view>::~compound_wrapper() at /usr/src/debug/scylla-3.0.rc3/keys.hh:165
 (inlined by) prefix_compound_wrapper<clustering_key_prefix, clustering_key_prefix_view, clustering_key_prefix>::~prefix_compound_wrapper() at /usr/src/debug/scylla-3.0.rc3/keys.hh:548
 (inlined by) clustering_key_prefix::~clustering_key_prefix() at /usr/src/debug/scylla-3.0.rc3/keys.hh:768
 (inlined by) range_bound<clustering_key_prefix>::~range_bound() at /usr/src/debug/scylla-3.0.rc3/./range.hh:35
 (inlined by) std::experimental::fundamentals_v1::_Optional_base<range_bound<clustering_key_prefix>, true>::~_Optional_base() at /opt/scylladb/include/c++/7/experimental/optional:290
 (inlined by) std::experimental::fundamentals_v1::optional<range_bound<clustering_key_prefix> >::~optional() at /opt/scylladb/include/c++/7/experimental/optional:493
 (inlined by) wrapping_range<clustering_key_prefix>::~wrapping_range() at /usr/src/debug/scylla-3.0.rc3/./range.hh:61
 (inlined by) nonwrapping_range<clustering_key_prefix>::~nonwrapping_range() at /usr/src/debug/scylla-3.0.rc3/./range.hh:430
 (inlined by) void std::_Destroy<nonwrapping_range<clustering_key_prefix> >(nonwrapping_range<clustering_key_prefix>*) at /opt/scylladb/include/c++/7/bits/stl_construct.h:98
 (inlined by) void std::_Destroy_aux<false>::__destroy<nonwrapping_range<clustering_key_prefix>*>(nonwrapping_range<clustering_key_prefix>*, nonwrapping_range<clustering_key_prefix>*) at /opt/scylladb/include/c++/7/bits/stl_construct.h:108
 (inlined by) void std::_Destroy<nonwrapping_range<clustering_key_prefix>*>(nonwrapping_range<clustering_key_prefix>*, nonwrapping_range<clustering_key_prefix>*) at /opt/scylladb/include/c++/7/bits/stl_construct.h:137
 (inlined by) void std::_Destroy<nonwrapping_range<clustering_key_prefix>*, nonwrapping_range<clustering_key_prefix> >(nonwrapping_range<clustering_key_prefix>*, nonwrapping_range<clustering_key_prefix>*, std::allocator<nonwrapping_range<clustering_key_prefix> >&) at /opt/scylladb/include/c++/7/bits/stl_construct.h:206
 (inlined by) std::vector<nonwrapping_range<clustering_key_prefix>, std::allocator<nonwrapping_range<clustering_key_prefix> > >::~vector() at /opt/scylladb/include/c++/7/bits/stl_vector.h:434
 (inlined by) query::partition_slice::~partition_slice() at /usr/src/debug/scylla-3.0.rc3/query.cc:106
std::default_delete<query::partition_slice const>::operator()(query::partition_slice const*) const at /opt/scylladb/include/c++/7/bits/unique_ptr.h:78
 (inlined by) std::unique_ptr<query::partition_slice const, std::default_delete<query::partition_slice const> >::~unique_ptr() at /opt/scylladb/include/c++/7/bits/unique_ptr.h:268
 (inlined by) query::querier<(emit_only_live_rows)1>::~querier() at /usr/src/debug/scylla-3.0.rc3/querier.hh:139
 (inlined by) void std::_Destroy<query::querier<(emit_only_live_rows)1> const>(query::querier<(emit_only_live_rows)1> const*) at /opt/scylladb/include/c++/7/bits/stl_construct.h:98
 (inlined by) void std::__detail::__variant::__erased_dtor<std::__detail::__variant::_Variant_storage<false, query::querier<(emit_only_live_rows)1>, query::querier<(emit_only_live_rows)0>, query::shard_mutation_querier> const&, 0ul>(std::__detail::__variant::_Variant_storage<false, query::querier<(emit_only_live_rows)1>, query::querier<(emit_only_live_rows)0>, query::shard_mutation_querier> const&) at /opt/scylladb/include/c++/7/variant:254
void std::__detail::__variant::_Variant_storage<false, query::querier<(emit_only_live_rows)1>, query::querier<(emit_only_live_rows)0>, query::shard_mutation_querier>::_M_reset_impl<0ul, 1ul, 2ul>(std::integer_sequence<unsigned long, 0ul, 1ul, 2ul>) at /opt/scylladb/include/c++/7/variant:351
 (inlined by) std::__detail::__variant::_Variant_storage<false, query::querier<(emit_only_live_rows)1>, query::querier<(emit_only_live_rows)0>, query::shard_mutation_querier>::_M_reset() at /opt/scylladb/include/c++/7/variant:356
 (inlined by) std::__detail::__variant::_Variant_storage<false, query::querier<(emit_only_live_rows)1>, query::querier<(emit_only_live_rows)0>, query::shard_mutation_querier>::~_Variant_storage() at /opt/scylladb/include/c++/7/variant:361
 (inlined by) std::__detail::__variant::_Variant_base<query::querier<(emit_only_live_rows)1>, query::querier<(emit_only_live_rows)0>, query::shard_mutation_querier>::~_Variant_base() at /opt/scylladb/include/c++/7/variant:390
 (inlined by) std::variant<query::querier<(emit_only_live_rows)1>, query::querier<(emit_only_live_rows)0>, query::shard_mutation_querier>::~variant() at /opt/scylladb/include/c++/7/variant:992
 (inlined by) query::querier_cache::entry::~entry() at /usr/src/debug/scylla-3.0.rc3/querier.hh:323
 (inlined by) void __gnu_cxx::new_allocator<std::_List_node<query::querier_cache::entry> >::destroy<query::querier_cache::entry>(query::querier_cache::entry*) at /opt/scylladb/include/c++/7/ext/new_allocator.h:140
 (inlined by) void std::allocator_traits<std::allocator<std::_List_node<query::querier_cache::entry> > >::destroy<query::querier_cache::entry>(std::allocator<std::_List_node<query::querier_cache::entry> >&, query::querier_cache::entry*) at /opt/scylladb/include/c++/7/bits/alloc_traits.h:487
 (inlined by) std::list<query::querier_cache::entry, std::allocator<query::querier_cache::entry> >::_M_erase(std::_List_iterator<query::querier_cache::entry>) at /opt/scylladb/include/c++/7/bits/stl_list.h:1815
 (inlined by) std::list<query::querier_cache::entry, std::allocator<query::querier_cache::entry> >::erase(std::_List_const_iterator<query::querier_cache::entry>) at /opt/scylladb/include/c++/7/bits/list.tcc:157
 (inlined by) query::querier_inactive_read::evict() at /usr/src/debug/scylla-3.0.rc3/querier.cc:239
reader_concurrency_semaphore::wait_admission(unsigned long, std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/reader_concurrency_semaphore.cc:80
_ZN27restricting_mutation_reader11with_readerIZNS_11fill_bufferENSt6chrono10time_pointIN7seastar12lowres_clockENS1_8durationIlSt5ratioILl1ELl1000EEEEEEEUlR20flat_mutation_readerE_EEDcT_S9_ at /usr/src/debug/scylla-3.0.rc3/mutation_reader.cc:604
 (inlined by) restricting_mutation_reader::fill_buffer(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/mutation_reader.cc:629
flat_mutation_reader::impl::operator()(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/flat_mutation_reader.hh:153
 (inlined by) flat_mutation_reader::operator()(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/flat_mutation_reader.hh:339
 (inlined by) cache::autoupdating_underlying_reader::move_to_next_partition(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/read_context.hh:78
range_populating_reader::operator()(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/row_cache.cc:536
 (inlined by) scanning_and_populating_reader::read_from_secondary(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/row_cache.cc:661
 (inlined by) scanning_and_populating_reader::read_from_primary(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#1}::operator()() const at /usr/src/debug/scylla-3.0.rc3/row_cache.cc:657
 (inlined by) seastar::apply_helper<scanning_and_populating_reader::read_from_primary(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#1}, std::tuple<>&&, std::integer_sequence<unsigned long> >::apply({lambda()#1}&&, std::tuple<>) at /usr/src/debug/scylla-3.0.rc3/seastar/core/apply.hh:35
 (inlined by) auto seastar::apply<scanning_and_populating_reader::read_from_primary(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#1}>(scanning_and_populating_reader::read_from_primary(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#1}&&, std::tuple<>&&) at /usr/src/debug/scylla-3.0.rc3/seastar/core/apply.hh:43
 (inlined by) seastar::future<seastar::optimized_optional<flat_mutation_reader> > seastar::futurize<seastar::future<seastar::optimized_optional<flat_mutation_reader> > >::apply<scanning_and_populating_reader::read_from_primary(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#1}>(scanning_and_populating_reader::read_from_primary(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#1}&&, std::tuple<>&&) at /usr/src/debug/scylla-3.0.rc3/seastar/core/future.hh:1389
 (inlined by) seastar::future<seastar::optimized_optional<flat_mutation_reader> > seastar::future<>::then<scanning_and_populating_reader::read_from_primary(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#1}, seastar::future<seastar::optimized_optional<flat_mutation_reader> > >(scanning_and_populating_reader::read_from_primary(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#1}&&) at /usr/src/debug/scylla-3.0.rc3/seastar/core/future.hh:952
 (inlined by) scanning_and_populating_reader::read_from_primary(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/row_cache.cc:657
scanning_and_populating_reader::read_next_partition(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/row_cache.cc:674
 (inlined by) scanning_and_populating_reader::fill_buffer(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#2}::operator()() const at /usr/src/debug/scylla-3.0.rc3/row_cache.cc:704
 (inlined by) seastar::future<> seastar::do_void_futurize_helper<seastar::future<> >::apply<scanning_and_populating_reader::fill_buffer(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#2}&>(scanning_and_populating_reader::fill_buffer(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#2}&) at /usr/src/debug/scylla-3.0.rc3/seastar/core/future.hh:1356
 (inlined by) seastar::future<> seastar::futurize<void>::apply<scanning_and_populating_reader::fill_buffer(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#2}&>(scanning_and_populating_reader::fill_buffer(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#2}&) at /usr/src/debug/scylla-3.0.rc3/seastar/core/future.hh:1382
 (inlined by) seastar::future<> seastar::do_until<scanning_and_populating_reader::fill_buffer(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#2}, scanning_and_populating_reader::fill_buffer(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#1}>(scanning_and_populating_reader::fill_buffer(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#1}, scanning_and_populating_reader::fill_buffer(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >)::{lambda()#2}) at /usr/src/debug/scylla-3.0.rc3/seastar/core/future-util.hh:502
scanning_and_populating_reader::fill_buffer(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/row_cache.cc:712
flat_mutation_reader::impl::operator()(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/flat_mutation_reader.hh:153
 (inlined by) flat_mutation_reader::operator()(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/flat_mutation_reader.hh:339
 (inlined by) operator() at /usr/src/debug/scylla-3.0.rc3/mutation_reader.cc:299
 (inlined by) apply<mutation_reader_merger::prepare_next(seastar::lowres_clock::time_point)::<lambda(mutation_reader_merger::reader_and_last_fragment_kind)>, mutation_reader_merger::reader_and_last_fragment_kind&> at /usr/src/debug/scylla-3.0.rc3/seastar/core/future.hh:1399
 (inlined by) futurize_apply<mutation_reader_merger::prepare_next(seastar::lowres_clock::time_point)::<lambda(mutation_reader_merger::reader_and_last_fragment_kind)>, mutation_reader_merger::reader_and_last_fragment_kind&> at /usr/src/debug/scylla-3.0.rc3/seastar/core/future.hh:1471
 (inlined by) parallel_for_each<__gnu_cxx::__normal_iterator<mutation_reader_merger::reader_and_last_fragment_kind*, std::vector<mutation_reader_merger::reader_and_last_fragment_kind> >, mutation_reader_merger::prepare_next(seastar::lowres_clock::time_point)::<lambda(mutation_reader_merger::reader_and_last_fragment_kind)> > at /usr/src/debug/scylla-3.0.rc3/seastar/core/future-util.hh:129
parallel_for_each<std::vector<mutation_reader_merger::reader_and_last_fragment_kind>&, mutation_reader_merger::prepare_next(seastar::lowres_clock::time_point)::<lambda(mutation_reader_merger::reader_and_last_fragment_kind)> > at /usr/src/debug/scylla-3.0.rc3/seastar/core/future-util.hh:175
 (inlined by) mutation_reader_merger::prepare_next(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/mutation_reader.cc:298
mutation_reader_merger::operator()(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/mutation_reader.cc:384
mutation_fragment_merger<mutation_reader_merger>::fetch(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/mutation_reader.cc:84
 (inlined by) mutation_fragment_merger<mutation_reader_merger>::operator()(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/mutation_reader.cc:115
 (inlined by) operator() at /usr/src/debug/scylla-3.0.rc3/mutation_reader.cc:475
repeat<combined_mutation_reader::fill_buffer(seastar::lowres_clock::time_point)::<lambda()> > at /usr/src/debug/scylla-3.0.rc3/seastar/core/future-util.hh:280
 (inlined by) combined_mutation_reader::fill_buffer(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/mutation_reader.cc:486
flat_mutation_reader::fill_buffer(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at /usr/src/debug/scylla-3.0.rc3/flat_mutation_reader.hh:383
 (inlined by) operator() at /usr/src/debug/scylla-3.0.rc3/mutation_reader.cc:828
 (inlined by) operator() at /usr/src/debug/scylla-3.0.rc3/mutation_reader.cc:753
operator() at /usr/src/debug/scylla-3.0.rc3/mutation_reader.cc:758
 (inlined by) apply<foreign_reader::forward_operation(seastar::lowres_clock::time_point, Operation) [with Operation = foreign_reader::fill_buffer(seastar::lowres_clock::time_point)::<lambda()>; Result = seastar::future<seastar::foreign_ptr<std::unique_ptr<seastar::circular_buffer<mutation_fragment, std::allocator<mutation_fragment> >, std::default_delete<seastar::circular_buffer<mutation_fragment, std::allocator<mutation_fragment> > > > >, bool>]::<lambda()>&> at /usr/src/debug/scylla-3.0.rc3/seastar/core/future.hh:1399
operator() at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.hh:344
apply at /usr/src/debug/scylla-3.0.rc3/seastar/core/apply.hh:35
 (inlined by) apply<seastar::smp_message_queue::async_work_item<Func>::process() [with Func = foreign_reader::forward_operation(seastar::lowres_clock::time_point, Operation) [with Operation = foreign_reader::fill_buffer(seastar::lowres_clock::time_point)::<lambda()>; Result = seastar::future<seastar::foreign_ptr<std::unique_ptr<seastar::circular_buffer<mutation_fragment, std::allocator<mutation_fragment> >, std::default_delete<seastar::circular_buffer<mutation_fragment, std::allocator<mutation_fragment> > > > >, bool>]::<lambda()>]::<lambda()>&> at /usr/src/debug/scylla-3.0.rc3/seastar/core/apply.hh:43
 (inlined by) apply_tuple<seastar::smp_message_queue::async_work_item<Func>::process() [with Func = foreign_reader::forward_operation(seastar::lowres_clock::time_point, Operation) [with Operation = foreign_reader::fill_buffer(seastar::lowres_clock::time_point)::<lambda()>; Result = seastar::future<seastar::foreign_ptr<std::unique_ptr<seastar::circular_buffer<mutation_fragment, std::allocator<mutation_fragment> >, std::default_delete<seastar::circular_buffer<mutation_fragment, std::allocator<mutation_fragment> > > > >, bool>]::<lambda()>]::<lambda()>&> at /usr/src/debug/scylla-3.0.rc3/seastar/core/future.hh:1343
 (inlined by) apply<seastar::smp_message_queue::async_work_item<Func>::process() [with Func = foreign_reader::forward_operation(seastar::lowres_clock::time_point, Operation) [with Operation = foreign_reader::fill_buffer(seastar::lowres_clock::time_point)::<lambda()>; Result = seastar::future<seastar::foreign_ptr<std::unique_ptr<seastar::circular_buffer<mutation_fragment, std::allocator<mutation_fragment> >, std::default_delete<seastar::circular_buffer<mutation_fragment, std::allocator<mutation_fragment> > > > >, bool>]::<lambda()>]::<lambda()>&> at /usr/src/debug/scylla-3.0.rc3/seastar/core/future.hh:1377
 (inlined by) operator() at /usr/src/debug/scylla-3.0.rc3/seastar/core/future-util.hh:86
 (inlined by) run_and_dispose at /usr/src/debug/scylla-3.0.rc3/seastar/core/task.hh:48
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:2695
seastar::reactor::run_some_tasks() at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:3118
seastar::reactor::run_some_tasks() at /usr/src/debug/scylla-3.0.rc3/seastar/util/log.hh:313
 (inlined by) seastar::reactor::run() at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:3265
seastar::app_template::run_deprecated(int, char**, std::function<void ()>&&) at /usr/src/debug/scylla-3.0.rc3/seastar/core/app-template.cc:185
main at /usr/src/debug/scylla-3.0.rc3/main.cc:854
__libc_start_main at ??:?
?? ??:0
```

```
2018-12-27 11:02:43,927 remote           L0194 DEBUG| [35.243.190.18] [stdout] Dec 27 11:02:43 info   |  [shard 2] compaction - Compacted 2 s
stables to [/var/lib/scylla/data/system/local-7ad54392bcdd35a684174e047860b377/mc-5106-big-Data.db:level=0, ]. 21952 bytes to 15688 (~71% of 
original) in 50ms = 0.30MB/s. ~256 total partitions merged to 1.
2018-12-27 11:02:44,015 remote           L0194 DEBUG| [35.196.94.126] [stdout] Dec 27 11:02:43 info   | Segmentation fault on shard 0.
2018-12-27 11:02:44,015 remote           L0194 DEBUG| [35.196.94.126] [stdout] Dec 27 11:02:43 info   | Backtrace:
 0x00000000006a7732 
 0x00000000005bee5c 
 0x00000000005bf105 
 0x00000000005bf153 
 /lib64/libpthread.so.0+0x000000000000f5cf 
 0x0000000002fc9962 
 0x0000000001347f3e 
 0x0000000003c0c6da 
 0x0000000003cceb25 
 0x00000000017a92f7 
 0x000000000152a991 
 0x000000000152e761 
 0x00000000015342da 
 0x000000000153572e 
 0x000000000179025d 
 0x00000000017928b7 
 0x0000000001793512 
 0x0000000001796808 
 0x0000000001797be6 
 0x00000000017658b6 
 0x0000000001767a69 
 0x0000000001768827 
 0x000000000176a8b3 
 0x00000000005a020e 
 0x00000000005a03d1 
 0x000000000066ea67 
 0x00000000007ac773 
 0x0000000000533957 
 /lib64/libc.so.6+0x00000000000223d4 
 0x000000000059a10d 

2018-12-27 11:03:24,339 threading        L0812 WARNI| Heartbeat failed for connection (140681860334672) to 35.196.94.126
2018-12-27 11:03:24,339 threading        L0812 DEBUG| Defuncting connection (140681860334672) to 35.196.94.126: errors=Connection heartbeat t
imeout after 30 seconds, last_host=35.196.94.126
2018-12-27 11:03:24,340 asyncorereactor  L0316 DEBUG| Closing connection (140681860334672) to 35.196.94.126
2018-12-27 11:03:24,340 asyncorereactor  L0320 DEBUG| Closed socket to 35.196.94.126
2018-12-27 11:03:24,340 threading        L0812 DEBUG| Defunct or closed connection (140681860334672) returned to pool, potentially marking host 35.196.94.126 as down
2018-12-27 11:03:24,341 thread           L0055 WARNI| Host 35.196.94.126 has been marked down
2018-12-27 11:03:24,341 thread           L0055 DEBUG| [control connection] Control connection host (35.196.94.126) is considered down, starting reconnection
2018-12-27 11:03:24,341 thread           L0055 DEBUG| [control connection] Attempting to reconnect
2018-12-27 11:03:24,341 thread           L0055 DEBUG| Removed connection pool for <Host: 35.196.94.126 datacenter1>
2018-12-27 11:03:24,342 thread           L0055 DEBUG| Starting reconnector for host 35.196.94.126
....
2018-12-27 11:03:43,776 thread           L0055 DEBUG| Error trying to reconnect control connection: NoHostAvailable('Unable to connect to any
 servers', {})
2018-12-27 11:03:43,911 output           L0642 DEBUG| SELECT * FROM ks_no_range_ghost_test.users errors={'35.196.94.126': 'Client request tim
eout. See Session.execute[_async](timeout)'}, last_host=35.196.94.126
2018-12-27 11:03:43,912 stacktrace       L0036 ERROR| 
2018-12-27 11:03:43,912 stacktrace       L0039 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado/core/test.py:449
2018-12-27 11:03:43,969 stacktrace       L0042 ERROR| Traceback (most recent call last):
2018-12-27 11:03:43,970 stacktrace       L0042 ERROR|   File "/home/amos/space/scylla-cluster-tests/upgrade_test.py", line 340, in test_rolli
ng_upgrade
2018-12-27 11:03:43,970 stacktrace       L0042 ERROR|     self.verify_db_data()
2018-12-27 11:03:43,970 stacktrace       L0042 ERROR|   File "/home/amos/space/scylla-cluster-tests/sdcm/fill_db_data.py", line 2930, in veri
fy_db_data
2018-12-27 11:03:43,970 stacktrace       L0042 ERROR|     self.run_db_queries(session, session.default_fetch_size)
2018-12-27 11:03:43,970 stacktrace       L0042 ERROR|   File "/home/amos/space/scylla-cluster-tests/sdcm/fill_db_data.py", line 2893, in run_
db_queries
2018-12-27 11:03:43,970 stacktrace       L0042 ERROR|     raise e
2018-12-27 11:03:43,970 stacktrace       L0042 ERROR| OperationTimedOut: errors={'35.196.94.126': 'Client request timeout. See Session.execut
e[_async](timeout)'}, last_host=35.196.94.126
....
Dec 27 11:04:28 rolling-upgrade-l-ak-30-db-node-fd01fdde-0-1 systemd[1]: scylla-server.service: main process exited, code=dumped, status=11/S
Dec 27 11:04:28 rolling-upgrade-l-ak-30-db-node-fd01fdde-0-1 systemd[1]: Unit scylla-server.service entered failed state. 
Dec 27 11:04:28 rolling-upgrade-l-ak-30-db-node-fd01fdde-0-1 systemd[1]: scylla-server.service failed. 
Dec 27 11:04:29 rolling-upgrade-l-ak-30-db-node-fd01fdde-0-1 systemd[1]: scylla-server.service holdoff time over, scheduling restart. 
Dec 27 11:04:29 rolling-upgrade-l-ak-30-db-node-fd01fdde-0-1 systemd[1]: Stopped Scylla Server. 
Dec 27 11:04:29 rolling-upgrade-l-ak-30-db-node-fd01fdde-0-1 systemd[1]: Starting Scylla Server... 
Dec 27 11:04:30 rolling-upgrade-l-ak-30-db-node-fd01fdde-0-1 scylla[12612]: Scylla version 3.0.rc3-0.20181224.5558fa8c4 starting ... 
````

**Comment by amoskong** [comment]
living monitor: 
http://35.243.229.117:3000/dashboard/db/scylla-per-server-metrics-nemesis-2-3?orgId=1&from=1545904796983&to=1545909531121

**Comment by denesb** [comment]
This appears to be the same problem as that in https://github.com/scylladb/scylla/issues/4018 (use-after-free on the semaphore evicting an inactive read). I will soon send a patch that fixes another possible place this might happen (Avi already sent a patch for one).

**Comment by denesb** [comment]
This issue can be closed. Fixed by 918d255 and e5a0ea3.
